### PR TITLE
TXAD-127: Implement filter to Quality Management Filters screen.

### PR DESCRIPTION
**Pull request by ruchawajpe**

### CODE REVIEW

**Review by Aditi-005 — APPROVED** [review]

File: lib/screens/qualityManagement/qm_list_screen.dart [hunk 1]
```diff
@@ -11,14 +11,18 @@ import '../../widgets/custom_icon_button_row.dart';
 import '../../widgets/custom_snackbar.dart';
 import '../../widgets/generic_app_bar.dart';
 import '../../widgets/generic_no_records_text.dart';
+import '../../widgets/text_button.dart';
 import 'qm_details_screen.dart';
+import 'qm_filters_screen.dart';
 import 'widgets/qm_list_tile_subtitle.dart';
 import 'widgets/qm_list_tile_titile.dart';
 
 class QualityManagementListScreen extends StatefulWidget {
```
Comment (objectorientguy) [inline lib/screens/qualityManagement/qm_list_screen.dart:20]: Why is it a stateful widget, and why is it approved @Aditi-005  ?

File: lib/screens/qualityManagement/widgets/qm_status_filter.dart [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import 'package:flutter/material.dart';
+import 'package:toolkit/configs/app_theme.dart';
+import 'package:toolkit/utils/database_utils.dart';
+
+import '../../../configs/app_color.dart';
+import '../../../configs/app_dimensions.dart';
+import '../../../data/enums/incident_and_qm_filter_status_enum.dart';
+
+class QualityManagementStatusFilter extends StatefulWidget {
```
Comment (objectorientguy) [inline lib/screens/qualityManagement/widgets/qm_status_filter.dart:9]: As mentioned above, why stateful?

**Review by objectorientguy — CHANGES_REQUESTED** [review]

File: lib/screens/qualityManagement/qm_list_screen.dart (hunk 1 above)
Comment (ruchawajpe) [reply lib/screens/qualityManagement/qm_list_screen.dart:20]: Pagination is implemented to this screen.

File: lib/screens/qualityManagement/widgets/qm_status_filter.dart (hunk 2 above)
Comment (ruchawajpe) [reply lib/screens/qualityManagement/widgets/qm_status_filter.dart]: Implemented it with bloc.

**Review by objectorientguy — APPROVED** [review]

**Pull request merged**